python/modules/IcePy/PySliceLoader.cpp

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) ZeroC, Inc.
+
+#include "PySliceLoader.h"
+#include "Types.h"
+
+#include <cassert>
+
+using namespace std;
+
+IcePy::PySliceLoader::PySliceLoader(PyObject* sliceLoader) : _sliceLoader(Py_NewRef(sliceLoader))
+{
+    assert(_sliceLoader != Py_None);
+
+    if (!PyCallable_Check(_sliceLoader))
+    {
+        throw Ice::InitializationException{__FILE__, __LINE__, "sliceLoader must be a callable"};
+    }
+}
+
+IcePy::PySliceLoader::~PySliceLoader() { Py_DECREF(_sliceLoader); }
```

Comment:
My understanding is this runs with the GIL locked, when communicator dealloc erases the map entry